sg14-error-codes.html

```diff
@@ -620,16 +658,26 @@
         </li>
     </ul>
     Arthur O'Dwyer's (admittedly uninformed) opinion is that these sound like antipatterns that could reasonably be avoided.
-    Niall Douglas's opinion seems to be either that these are valid patterns, or else that they turn out to be impossible to
-    avoid in practice. Charley Bay has also seen these failure modes in practice.
+    Niall Douglas's opinion is that these are patterns in use in big consumers e.g. Python loads its C/C++ modules with
+    <code>RTLD_LOCAL</code>, and indeed has no choice but to do so. Charley Bay has also seen these failure modes in practice.
 </p>
 <p>
-    The practical difficulty of using singletons in DLLs seems to be a continuing pain point for certain programmers. The use
+    The practical difficulty of using singletons in header only libraries seems to be a continuing pain point for certain programmers. The use
     of singletons by RTTI seems to be a continuing reason that some programmers avoid <code>dynamic_cast</code> and
     exception handling. The use of singletons by <code>std::error_category</code> will cause the same kinds of problems in practice
     as the use of singletons by RTTI in exception-handling. If we cannot figure out how to eliminate these practical problems,
     then <code>std::error_code</code> cannot possibly be a suitable replacement for exception-handling because it will
     continue to have the same problems (namely, a reliance on singletons compared for address-equality).
+    
+    <!-- You are aware right that C++ exception throws have the exact same singleton problem too? The EH runtime uses
+    the vtable when catching exception throws, and it assumes that the vtable is unique per unique type thrown. Which
+    is not the case in a multiple -fvisibility=hidden use case on ELF. ELF is so profoundly broken for C++ it's unreal, at
+    least MachO and PE have viable workarounds for this crap. On ELF you are literally flying constantly on empty.
```

Comment:
This does nothing to change Arthur's (admittedly uninformed) opinion that all these problems are caused by misuse of `RTLD_LOCAL` and if you just stopped doing that you wouldn't have the problems. ;)  We will not be able to settle this in GitHub comments though.